src/mame/drivers/krz2000.cpp

```diff
@@ -52,63 +65,315 @@ class k2000_state : public driver_device
+	// 	0xFFFC02 = 0x00
+	//  0xFFFC03 = 0x30
+	//   For CS1:
+	//  0xFFFC04 = 0x60 = 0b0110 0000 = Start address of the memory area for CS1 = A21 + A22 
+	//  0xFFFC05 = 0x3F = 0b0011 1111 = Size of memory area = M19 + M18 + M17 + M16 + M15-M9 + M8 = 1MB
+	//  0xFFFC06 = 0x00
+	//  0xFFFC07 = 0x30 = 0b0011 0000 = Area enabled, use external DTACK
+	// By this logic, Hobbes chips are mapped to 0x100000, 0x200000, 0x400000, 0x600000
+	// And their address mask is 0x000FFF00, which means the effective ranges are:
+	// 0x100???00
+
+	// Hobbes chip selection seems to look at CA19 and CS1
+
+	// Internal sample ROM is switched on SBA20,SBA21, which corresponds to CA21,CA22 (one bit off)
+	// meaning the internal address space for accessing the sample ROM is 0x200000 through 0x6fffff (?)
+	// Blocks 2, 3, 4, 5, 6
+	// map(0x200000, 0x2fffff).rom().region("pcm", 0);		  // Block 2: Sample ROM
+	// map(0x300000, 0x3fffff).rom().region("pcm", 0x100000); // Block 3: Sample ROM
+	// map(0x400000, 0x4fffff).rom().region("pcm", 0x200000); // Block 4: Sample ROM
+	// Writes to 0x4FFD82, 0x4FFD88, 0x4FFE24
+
+	// Hypothesis: Calvin (or Janis) is mapped at 0x500000
+	//  Writes to 0x500402, 0x500418, 0x50041A early in the boot process
+	//  Initialization then writes zeros in a descending ladder:
+	//  0x500400, 0x500402
+	//  0x5003FC, 0x5003FE, ...
+	//  down to 0x500004 (then 0x500006)
+
+	// Calvin memory?
+	map(0x500000, 0x5003FF).rw(FUNC(k2000_state::calvin_read), FUNC(k2000_state::calvin_write));
+	// Calvin control registers?
+	map(0x500400, 0x5004FF).rw(FUNC(k2000_state::calvin_read), FUNC(k2000_state::calvin_write));
+	// Reads and writes to 0x580000
+
+	// Hobbes (it seems) is connected in the 0x600000 range
+	// The second Hobbes chip (?) seems connected at 0x680000
+	// Initialization seems to:
+	//  - write to 0x600300, followed by 0x600xxx where xxx is in 0x400 through 0x5FF
+	//  - write to 0x6?0002, 0x6?0006, 0x6?0004, through 0x6?000F4
+	//  - write to 0x600004, 0x600014, 0x600024... through 0x6002F4
+	//    (with a repeating pattern of [0x02, 0x00, 0x26, 0x24, 0x4A, 0x48])
+	//  - write to 0x680004, 0x680014, 0x680024... through 0x6802F4
+	//    (with a repeating pattern of [0x80, 0x02, 0xA4, 0x26, 0xC6, 0x4A])
+	// Each Hobbes chip seems to have two control registers, mapped at 0x600300/0x680300 and 0x600400/0x680400
+
+	// Hobbes memory?
+	map(0x600000, 0x6002FF).rw(FUNC(k2000_state::hobbes0_read), FUNC(k2000_state::hobbes0_write));
+	// Hobbes control registers?
+	map(0x600300, 0x6005FF).rw(FUNC(k2000_state::hobbes0_read), FUNC(k2000_state::hobbes0_write));
+
+	// Hobbes memory?
+	map(0x680000, 0x6802FF).rw(FUNC(k2000_state::hobbes1_read), FUNC(k2000_state::hobbes1_write));
+	// Hobbes control registers?
+	map(0x680300, 0x6805FF).rw(FUNC(k2000_state::hobbes1_read), FUNC(k2000_state::hobbes1_write));
+
+	// SROM - Setup ROM?
+	uint32 srom_base = 0x280000;
+	map(srom_base, srom_base + 0x40000 - 1).rom().region("maincpu", 0x100000);
```

Comment:
imho calculations and notes aren't a thing for mapping. They should be as simple as possible so that the next guy debugging this doesn't struggle in understand what a given access should map to.
For notes, my suggestion is that if you need more than single liners then shouldn't be inside address_map at all but moved to a separate notes section, either near the relevant actual implementation detail or in file header. Also cfr. the floppy section just below this, where you repeat the mapping note twice.